File: CS214-Programming Language Concepts/projects/06/ruby/array.rb
# array.rb arrays.cpp reads in a user-specified amount of values into an array, then prints them out. Driver driven.  # Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Joshua Maguire
# Date:4/2/14
##############################################


###################################
#Read Array reads in contents of array
#input: an array and its size
##############################3
def readArray(theArray, itsSize)
	print "\nEnter arrays contents:\n"
	#theArray.each { |num| num=gets.chomp.to_i} Why doesn't this work?
	while itsSize > 0 do
		theArray.push(gets.chomp.to_i)
		itsSize=itsSize-1
	end
end
###############################
#printArray simply puts the arrays contents
##############################
def printArray (theArray)
 puts theArray
end

#main
if __FILE__ == $0
   print "\nEnter array size:"
   arraySize=gets.chomp.to_i
   anArray=Array.new(arraySize)
   readArray(anArray, arraySize)
   printArray(anArray)
end

File: CS112-Intro to Datastructures/lab2/array.cpp
/* array.cpp defines "C style" array operations
 * Name:Joshua Maguire
 * Date:2/11/13
 * Begun by: Joel Adams, for CS 112 at Calvin College.
 */
 
#include "array.h"

void initialize(double *a, int size) {
	int val = 0;
	for (int i = 0; i < size; i++) {
		val = i+1;
		a[i] = val;
	}
}
 
void print(double *a, int size) {
	for (int i = 0; i < size; i++) {
		cout << *a << '\t';
		a++;
	}
}		

double average(double *a, int size) {
	double result;
	for (int i = 0; i < size; i++) {
		result+=a[i];
	}
	result=result/size;
	return result;
}
